racket/collects/racket/contract/private/arr-i.rkt

```diff
@@ -545,6 +573,7 @@ evaluted left-to-right.)
          (for/list ([arg (in-list args)])
            (hash-ref wrapper-args arg)))
        #`(return/no-unsupplied #,(if (null? result-checkers) #f (car result-checkers))
+                               (list #,@param-mark)
                                #,(if rst
                                      #'rest-args
                                      #''())
```

Comment:
I don't know the performance implication of these additional lists (probably low) but this isn't no change to the generated code, right?

Reply:
Ah yes, that's a good catch. An easy solution is to create a variant, `return/no-unsupplied*`, with the extra argument and then expand to that version when needed. I could probably abstract over both versions with a macro to avoid code duplication. Does that sound reasonable?

Reply:
I haven't been using macros for abstraction like that in this code so I'm unsure about that aspect.

As for the name, the words in the name are intended to say what case it is, so probably another slash with another word is better than a star suffix.